ChemistryLib/PhreeqcInterface.cpp

```diff
@@ -0,0 +1,312 @@
+             ++item_id)
+        {
+            auto const& accepted_item = output.accepted_items[item_id];
+            auto const& item_type = accepted_item.getItemType();
+            auto const& item_name = accepted_item.getName();
+            switch (item_type)
+            {
+                case ItemType::pH:
+                {
+                    // Update pH value
+                    solution.setpH(accepted_items[item_id]);
+                    break;
+                }
+                case ItemType::pe:
+                {
+                    // Update pe value
+                    solution.setpe(accepted_items[item_id]);
+                    break;
+                }
+                case ItemType::Component:
+                {
+                    // Update component concentrations
+                    auto& component = BaseLib::findElementOrError(
+                        components.begin(), components.end(),
+                        [&item_name](Component const& component) {
+                            return component._name == item_name;
+                        },
+                        "Could not find component '" + item_name + "'.");
+                    component._amount = accepted_items[item_id];
+                    break;
+                }
+                case ItemType::EquilibriumPhase:
+                {
+                    // Update amounts of equilibrium phases
+                    auto& equilibrium_phase = BaseLib::findElementOrError(
+                        nodal_equilibrium_phases.begin(),
+                        nodal_equilibrium_phases.end(),
+                        [&item_name](
+                            EquilibriumPhase const& equilibrium_phase) {
+                            return equilibrium_phase._name == item_name;
+                        },
+                        "Could not find equilibrium phase '" + item_name +
+                            "'.");
+                    equilibrium_phase._amount = accepted_items[item_id];
+                    break;
+                }
+                case ItemType::KineticReactant:
+                {
+                    // Update amounts of kinetic reactants
+                    auto& kinetic_reactant = BaseLib::findElementOrError(
+                        nodal_kinetic_reactants.begin(),
+                        nodal_kinetic_reactants.end(),
+                        [&item_name](KineticReactant const& kinetic_reactant) {
+                            return kinetic_reactant._name == item_name;
+                        },
+                        "Could not find kinetic reactant '" + item_name + "'.");
+                    kinetic_reactant._amount = accepted_items[item_id];
+                    break;
+                }
+                case ItemType::Unknown:
+                {
+                    OGS_FATAL(
+                        "Unexpected exception occurred in reading phreeqc "
+                        "outputs.");
```

Comment:
One could provide useful information here, like what was read and when, and into what type of unknown value it run, and what one could do about it.
This is a normal error handling path, not an unexpected exception.

Reply:
I see. Removed, because such a situation could be prevented with `ChemistryLib::Output`.